pydantic/schema.py

```diff
@@ -831,10 +833,10 @@ def go(type_: Any) -> Type[Any]:
         attrs: Optional[Tuple[str, ...]] = None
         constraint_func: Optional[Callable[..., type]] = None
         if isinstance(type_, type):
-            if issubclass(type_, str) and not issubclass(type_, (EmailStr, AnyUrl, ConstrainedStr)):
+            if issubclass(type_, (str, StrLike)) and not issubclass(type_, (EmailStr, AnyUrl, ConstrainedStr)):
```

Comment:
I'm not really in favor of those two new classes `BytesLike` and `StrLike` because I would expect `EmailStr`, `ConstrainedStr` and so on to inherit from `StrLike`.
I would simply write `if issubclass(type_, (str, SecretStr))` and `elif issubclass(type_, (bytes, SecretBytes))`
and remove the two new classes.
It's just my opinion of course but I reckon it's confusing

Reply:
I added those classes for the future extending.

For instance, if there will be some new classes which would not be str, but should be validated as str, like `SecretStr` and `SecretBytes`, you can simply inherit from the marker class. 

`EmailStr`, `AnyUrl`, `ConstrainedStr` those classes actually inherits from `str` so they are strings. But `secret` classes they aren't strings but they should be validated as strings, that's why I don't inherit `EmailStr`, `AnyUrl`, `ConstrainedStr` classes from `StrLike`.

@patrickkwang what is your opinion, should I remove `StrLike` and `BytesLike` classes?

Reply:
I tend to prefer adding "genericity" later on. Since we only have one class and I'm not convinced we'll add new ones for a while, I would remove them yes. If some day we have something like `issubclass(type_, (str, SecretStr, C1, C2))` then it won't cost much to do what you did here and add a dummy parent class. WDYT ?

Reply:
I totally agree with you, for this moment it adds more complexity to solve a simple problem.

But in the future, we should definitely think about some "genericity"😊.

I will remove those classes and update PR.
Thanks for a review.